### ISSUE TITLE
Invisible weapons. Bug ragdoll animations. September 2019, PublicPreview Build, the problems with import mixamo characters.

### ISSUE THREAD

**Issue by janekste** [TheGameCreators/GameGuruRepo#614]
First time here write. English is not my native language, so sorry if there are mistakes in the text.

Hey. Performance Update great. I don't know if it's a bug or not, but FBX mixamo characters are holding invisible weapons. To create a character, I used the FUSE editor, which is available in the steam store. Import works. Everything's fine. Only weapons invisible in the hands of, if his occasionally in FPE file. 

Also ragdoll animations do not work. If the imported FBX mixamo character to kill, then its texture spreads across locations. If ragdoll off, then everything works fine. Think this bug. I hardly did anything wrong. But we can't rule out a mistake on my part. Screenshots with the problem attached.
![невидимое оружие](https://user-images.githubusercontent.com/55485525/65135353-77740400-da0e-11e9-9788-7a936e38a19c.png)
![невидимое оружие2](https://user-images.githubusercontent.com/55485525/65135354-77740400-da0e-11e9-8af3-cc4c0129847b.png)
![смерть рагдолл 1](https://user-images.githubusercontent.com/55485525/65135358-78a53100-da0e-11e9-903c-4b50ee475b60.png)
![смерть рагдолл 0](https://user-images.githubusercontent.com/55485525/65135360-793dc780-da0e-11e9-9dbc-09b16c0d3500.png)
![fpe settings](https://user-images.githubusercontent.com/55485525/65136230-e3a33780-da0f-11e9-9d09-249134094aee.png)

Today (28 January 2020) I decided to reproduce the error again. All errors are in place.

**Comment by synchromesh62** [comment]
@LeeBamberTGC 
@plemsoft 
I am unable to test this but this may have been fixed with  the latest importer ?

**Comment by janekste** [comment]
![image](https://user-images.githubusercontent.com/55485525/74463310-55d25800-4e9a-11ea-9b5b-4d54c4fd6e07.png)
![image](https://user-images.githubusercontent.com/55485525/74463334-5ec32980-4e9a-11ea-9e8a-1ec04c8ed8b6.png)
![image](https://user-images.githubusercontent.com/55485525/74463347-6387dd80-4e9a-11ea-918c-7ba418f0fe3e.png)
![image](https://user-images.githubusercontent.com/55485525/74463366-6a165500-4e9a-11ea-828e-fc525b3e37e2.png)
![image](https://user-images.githubusercontent.com/55485525/74463383-700c3600-4e9a-11ea-91ad-86b96b10eee8.png)
Here are pictures of all the steps. I attached the character model file, its FPE file, and textures.
[trup_v_maske.zip](https://github.com/TheGameCreators/GameGuruRepo/files/4200089/trup_v_maske.zip)
I also attached the character model in T-pose to make it easier for you to check everything.
[T-pose_character.zip](https://github.com/TheGameCreators/GameGuruRepo/files/4200120/T-pose_character.zip)

**Comment by plemsoft** [comment]
@janekste Thanks for posting your issue,

Direct import of mixamo character is not yet complete , it do support some conversion of bone names to gg bone names , unfortunately the names in FBX files is not the same as GG currently support , if you export your model as .x the bone names will be converted into something GG understand. You should try that first :)

Also there is many threads on the forum about how to import mixamo character , you could try to read those and see if that help you.

I will mark this as an enhancement as direct FBX mixamo import is not currently supported :)

**Comment by janekste** [comment]
@plemsoft Not completed? What about the September 2019 update? It says "Improved MIXAMO FBX Importing (2-chain-finger in T-POSE), simply select 'Uber' in 'Is Character Field'".
https://www.game-guru.com/news-post/gameguru-september-update-released

**This is an ERROR, not an enhancement.**

It is not good that the added function is not completed after 7 months after the update. You are working very slowly, developers. Very very slow!  I hope it will take you less than 5 years to fix this.

Thank you for your explanation. But it's not what I expected.
![image](https://user-images.githubusercontent.com/55485525/77789639-0ae85a80-706c-11ea-806a-7a48a343adf5.png)

**Comment by plemsoft** [comment]
@janekste Sorry to hear ,

Anyway the improved mixamo import was not for chars with guns , you still need to add the "firespot" bone yourself before you import in GameGuru. 

Also ragdoll do not work with all mixamo chars, you can disable ragdoll in the fpe "ragdoll = 0" , and add a dead animation instead "anim6 = 4800,4958" , this is the only way at the moment, hope this explains it :)

**Comment by LeeBamberTGC** [comment]
As Preben says, the update did improve the importing of character a lot, but you are still going to be bound by a very specific pipeline to get the character in and working as intended.

There is no support for automatic weapon config on the imported character, no support for ragdoll on Mixamo bone structures and the character FPE needs to be manually edited to finish the character configuration. 

For Mixamo, we recommend:

;identity details
ischaracter      = 1
hasweapon        = modern\colt1911
handfirespotoffx = 0.0707
handfirespotoffy = -2.3718
handfirespotoffz = -1.6504
handfirespotrotx = -44.2759
handfirespotroty = -22.4883
handfirespotrotz = -26.3792
handfirespotsize = 100
cantakeweapon    = 1
ragdoll          = 0

;ai
aimain           = ai_soldier.lua

;anim
animmax          = 5
playanimineditor = 1
ignorecsirefs    = 1
anim0            = 100,205   ;idle
anim1            = 687,707   ;move
anim2            = 5511,5553 ;kick
anim3            = 4812,4850 ;hurt
anim4            = 515,605   ;reload

As this was bug fixing week, and we hit our goal of fixing every bug reported since the end of Feb 2020, I decided to stay up a little late and add some extra functionality. When you import a character without a firespot, it will look for the Bip01_R_Hand limb name instead and add a firespot,

And just so flexibility can be maintained, I have added new FPE fields starting HANDFIRESPOT which can adjust the placement of the weapon within the right hand of the character:

handfirespotoffx = 0.0707
handfirespotoffy = -2.3718
handfirespotoffz = -1.6504
handfirespotrotx = -44.2759
handfirespotroty = -22.4883
handfirespotrotz = -26.3792
handfirespotsize = 100
